CRM/ManualDirectDebit/Hook/PostProcess/Membership/DirectDebitMandate.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Class provide fetching Direct Debit Mandate Data from Membership form
+ * and launch saving in Database
+ */
+class CRM_ManualDirectDebit_Hook_PostProcess_Membership_DirectDebitMandate {
+
+  /**
+   * Form object that is being altered.
+   *
+   * @var object
+   */
+  private $form;
+
+  /**
+   * List Of mandate custom group fields
+   *
+   * @var array
+   */
+  private $listOfDirectDebitMandateCustomGroupFields;
+
+  /**
+   * List of mandate date
+   *
+   * @var array
+   */
+  private $mandateValues;
+
+  /**
+   * Id of current mandate
+   *
+   * @var int
+   */
+  private $currentContactId;
+
+  public function __construct($form) {
+    $this->form = $form;
+    $this->currentContactId = $this->form->getVar('_contactID');
+
+    $mandateDataProvider = new CRM_ManualDirectDebit_Common_DirectDebitDataProvider();
+    $this->listOfDirectDebitMandateCustomGroupFields = $mandateDataProvider->getMandateCustomFieldNames();
+  }
+
+  /**
+   * Fetches Direct Debit Mandate Data
+   */
+  public function fetchMandateData() {
```

Comment:
I don't think this method really fetches mandate data. If I understand correctly, this method stores the data, so maybe that would be a better name, `storeMandateData`.  Or `saveMandateData` may be better still.